src/python/pants/bin/local_pants_runner.py

```diff
@@ -197,6 +199,32 @@ def _perform_run_body(self, goals: tuple[str, ...], poll: bool) -> ExitCode:
     def _finish_run(self, code: ExitCode) -> None:
         """Cleans up the run tracker."""
 
+    def _maybe_report_memory_summary(self) -> None:
+        global_options = self.options.for_global_scope()
+        if not global_options.memory_summary:
+            return
+
+        ids: set[int] = set()
+        count_by_type: Counter[type] = Counter()
+        sizes_by_type: Counter[type] = Counter()
+
+        items, rust_sizes = self.graph_session.scheduler_session.live_items()
+        for item in items:
+            count_by_type[type(item)] += 1
+            sizes_by_type[type(item)] += deep_getsizeof(item, ids)
+
+        entries = [
+            (size, count_by_type[typ], f"{typ.__module__}.{typ.__qualname__}")
+            for typ, size in sizes_by_type.items()
+        ]
+        entries.extend(
+            (size, count, f"(native) {name}") for name, (count, size) in rust_sizes.items()
+        )
+
+        print("Memory summary:", file=sys.stderr)
```

Comment:
It'd be good to include what each column's title is.

Reply:
Agreed... either here or in the help.